extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -31,18 +32,18 @@ export class WorkspaceService implements IWorkspaceService {
 	}
 
 	/**
-	 * Verify that a workspace is open or that if one isn't, it's ok to create a workspace and restart ADS
+	 * Verify that a workspace is open or that if one isn't and we're running in ADS, it's ok to create a workspace and restart ADS
 	 */
 	async validateWorkspace(): Promise<boolean> {
-		if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length === 0) {
+		if (getAzdataApi() && (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length === 0)) {
 			const result = await vscode.window.showWarningMessage(constants.RestartConfirmation, { modal: true }, constants.OkButtonText);
 			if (result === constants.OkButtonText) {
 				return true;
 			} else {
 				return false;
 			}
 		} else {
-			// workspace is open
+			// workspace is open or we're running in vs code
```

Comment:
shouldn't we validate the workspace here also before returning true?

Reply:
For what? VS Code? No - it's only something we do in ADS because we require reloading the window when a file is the first one being added to a workspace. In VS Code they just restart the extension host - but that happens invisibly to the user and so there's no need to have this warning.

Reply:
I added another comment to more clearly explain this